### PR TITLE
fix: return empty result from git log on repo with no commits

### LINKED ISSUE #1155: Handle empty list of commits in new repository
I started to create a PR for this issue as it is simple and then I saw in `CONTRIBUTING.md`

> ... it does not introduce unnecessary abstraction layers or modify Git's core functionality.

so I think that raising an issue to ask is more sensible.

My first attempt with this gem was to try

```ruby
repo = Git.open(File.expand_path('source', __dir__))
repo.log.execute
```

on a new and completely empty repository. This results in an exception being thrown as `git log` exits with a fatal error.

```bash
$ git log
fatal: your current branch 'master' does not have any commits yet
```

 I would expect this to fail silently and return an empty array but I appreciate that this differs from the functionality of `git`. Is this a change that you would consider?

### PR DESCRIPTION
**Pull request by jcouball**
## Summary

Port of #1159 (4.x branch) to the `main` branch.

When `git log` is run against an unborn branch (a branch with no commits), return an empty array instead of raising an error.

## Changes

- Add `log_or_empty_on_unborn` helper in `Git::Lib` that rescues `Git::FailedError` when exit status is 128 and stderr matches "does not have any commits yet"
- Wrap `log_commits` and `full_log_commits` with this helper to return `[]` on empty repos
- Add unit tests for both `Git::Lib#log_commits` and `Git::Lib#full_log_commits` on empty repos

Fixes #1155

### CODE REVIEW

File: lib/git/lib.rb [hunk 1]
```diff
@@ -341,17 +341,16 @@ def full_log_commits(opts = {})
...
-      )
-      process_commit_log_data(result.stdout.split("\n"))
+      # args = log_common_options(opts)
+      # args += build_args(opts, FULL_LOG_EXTRA_OPTIONS_MAP)
+      # args += log_path_options(opts)
+
+      # log_or_empty_on_unborn do
+      #   full_log = command_lines('log', *args)
+      #   process_commit_log_data(full_log)
+      # end
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/lib.rb]: Remove the comments.
